modules/core/src/main/java/org/apache/ignite/internal/processors/query/schema/SchemaIndexCacheVisitorImpl.java

```diff
@@ -155,28 +223,35 @@ public SchemaIndexCacheVisitorImpl(GridCacheContext cctx, SchemaIndexOperationCa
      * @param parts Partitions.
      * @param clo Closure.
      * @param remainder Remainder.
+     * @return Index statistics.
      * @throws IgniteCheckedException If failed.
      */
-    private void processPartitions(List<GridDhtLocalPartition> parts, SchemaIndexCacheVisitorClosure clo,
+    private SchemaIndexCacheStat processPartitions(List<GridDhtLocalPartition> parts, SchemaIndexCacheVisitorClosure clo,
         int remainder)
         throws IgniteCheckedException {
+
+        SchemaIndexCacheStat tmp = new SchemaIndexCacheStat();
```

Comment:
I'd suggest to return null and not collect any stats if the new sys prop is false.

Reply:
done